cpp/src/arrow/compute/api_aggregate.h

```diff
@@ -46,7 +46,7 @@ class ExecContext;
 class ARROW_EXPORT ScalarAggregateOptions : public FunctionOptions {
  public:
   explicit ScalarAggregateOptions(bool skip_nulls = true, uint32_t min_count = 1);
-  constexpr static char const kTypeName[] = "ScalarAggregateOptions";
+  static constexpr char const kTypeName[] = "ScalarAggregateOptions";
```

Comment:
This is working around a slightly silly parsing bug: https://github.com/michaeljones/breathe/issues/787

@lidavidm What do you think?

Reply:
I think it's fine, though it would be nice to get it fixed upstream eventually.